management/virtualmachine/client.go

```diff
@@ -265,7 +266,7 @@ func (vm VirtualMachineClient) DeleteRole(cloudServiceName, deploymentName, role
 		return "", fmt.Errorf(errParamNotSpecified, "roleName")
 	}
 
-	requestURL := fmt.Sprintf(azureRoleURL, cloudServiceName, deploymentName, roleName)
+	requestURL := fmt.Sprintf(azureRoleURLDelete, cloudServiceName, deploymentName, roleName)
```

Comment:
@kbxkb Thanks for your pull request. At this line, you're always deleting the disks implicitly. I don't think this is very desirable. I prefer adding a `bool deleteDisks` parameter and appending `?comp=media` to the URL here.